eth-contracts/scripts/migrate-contracts.js

```diff
@@ -82,27 +82,83 @@ const outputJsonConfigFile = async (outputFilePath) => {
+ */
+ const outputFlaskConfigFile = async (outputPath) => {
+  try {
+    // Pull registry address from config because artifacts will require the updated
+    // version after a migration resuses the registry
+    let migrationOutputPath = path.join(getDirectoryRoot(AudiusEthContracts), 'migrations', 'migration-output.json')
+    if (!fs.existsSync(migrationOutputPath)) {
+      console.log('Failed to find migration output')
+      throw new Error('Failed to find migration output')
+    }
+    const addressInfo = require(migrationOutputPath)
+
+    let configFileContents = '[eth_contracts]\n'
+    configFileContents += 'registry = ' + addressInfo.registryAddress + '\n'
+
+    configFileContents += '\n'
+
+    let outputFlaskConfigFile = outputPath
+    console.log(`Target Output Flask Config File: ${outputFlaskConfigFile}`)
+    console.log(`Contents: \n ${configFileContents}`)
+
+    fs.writeFile(outputFlaskConfigFile, configFileContents, err => {
+      // throws an error, you could also catch it here
+      if (err) throw err
+
+      // success case, the file was saved
+      console.log(`Environment file written: ${outputFlaskConfigFile}`)
+    })
+  } catch (e) {
+    console.log(e)
+  }
+}
+
 /** Replace eth-contracts artifacts in libs with new ABIs and config */
 module.exports = async callback => {
-  const libsDirRoot = path.join(getDirectoryRoot(Libs), 'eth-contracts')
-  console.log(libsDirRoot)
-  fs.removeSync(libsDirRoot)
-
-  await copyBuildDirectory(path.join(libsDirRoot, '/ABIs'))
-  await outputJsonConfigFile(path.join(libsDirRoot, '/config.json'))
+  // output to Libs
+  try {
+    const libsDirRoot = path.join(getDirectoryRoot(Libs), 'eth-contracts')
+    fs.removeSync(libsDirRoot)
+  
+    await copyBuildDirectory(path.join(libsDirRoot, '/ABIs'))
+    await outputJsonConfigFile(path.join(libsDirRoot, '/config.json'))
+  } catch (e) {
+    console.log("Libs doesn't exist", e)
+  }
 
   // output to Identity Service
   try {
     await outputJsonConfigFile(path.join(getDirectoryRoot(AudiusIdentityService), '/eth-contract-config.json'))
   } catch (e) {
-    console.log("Identity service doesn't exist, probably running via E2E setup scripts", e)
+    console.log("Identity service doesn't exist", e)
   }
 
-  // output to Creator Node
+  // output to Content Node
   try {
     await outputJsonConfigFile(path.join(getDirectoryRoot(AudiusCreatorNode), '/eth-contract-config.json'))
   } catch (e) {
-    console.log("Creator node doesn't exist, probably running via E2E setup scripts", e)
+    console.log("Creator node doesn't exist", e)
+  }
+
+  // output to Discovery Node
+  try {
+    const discProvOutputPath = path.join(getDirectoryRoot(AudiusDiscoveryNode), 'build', 'eth-contracts')
+
+    // Copy build directory
+    await copyBuildDirectory(discProvOutputPath)
+
+    const flaskConfigPath = path.join(
+      getDirectoryRoot(AudiusDiscoveryNode),
+      'eth_contract_config.ini'
```

Comment:
thanks for making this consistent, tbh tho i do think it would be cleaner to not have either contract_config.ini or eth_contract_config.ini and simply manipulate the `.env` with just `default_config.ini` not a blocker

Reply:
yes i agree, just didn't want to change too too much here

Reply:
prior to this change, discovery was just ignoring the deployed eth contracts b/c they weren't needed for tests 😬